src/proxy/http/dialect.rs

```diff
@@ -0,0 +1,115 @@
+use http;
+
+use proxy::http::h1;
+
+/// Describes the required HTTP "dialect".
+///
+/// This marks whether to use HTTP/2 or HTTP/1.x for a request. In
+/// the case of HTTP/1.x requests, it also stores a "host" key to ensure
+/// that each host receives its own connection.
```

Comment:
Besides the "host" key other attributes are stored, so this documentation is misleading by omission.

It seems like this type is used as part of the key that determines what connection to route a request to, based on the HTTP version and other version-related properties of the request. I think the documentation should say something like that.